### ISSUE TITLE
Javadoc: type attribute of ordered lists not preserved (in xml)

### ISSUE THREAD

**Issue by Stewori** [doxygen/doxygen#8522]
**Describe the bug**
Processing Java code
```
/**
 * Doxygen does not preserve the type attribute of ordered lists:
 *
 * <ol type="a">
 *   <li>an item</li>
 *   <li>another item</li>
 *   <li>yet another item</li>
 * </ol>
 */
public class Test {}
```
results in the following xml:
```
...
<orderedlist>
<listitem>
<para>an item </para>
</listitem>
<listitem>
<para>another item </para>
</listitem>
<listitem>
<para>yet another item </para>
</listitem>
</orderedlist>
...
```
It is now impossible to know that the author intended alphabetical numbering. This can become an issue if the text references list items by their letter rather than by number.

**Expected behavior**
I expect the xml output to preserve the type attribute of html's ``ol`` tag, e.g. like this
```
...
<orderedlist type="a">
<listitem>
<para>an item </para>
</listitem>
<listitem>
<para>another item </para>
</listitem>
<listitem>
<para>yet another item </para>
</listitem>
</orderedlist>
...
```

**To Reproduce**
Find a Java file and doxygen config attached. The issue is demonstrated like mentioned above.
After running doxygen on the example, the files in the subfolder ``xml`` contain the snippet shown above.
[issue_javadoc_ol_type.zip](https://github.com/doxygen/doxygen/files/6388164/issue_javadoc_ol_type.zip)

**Version**
Tested with doxygen clone from 2021-04-11:
```
/blah/doxygen/build/bin/doxygen -v
1.9.2 (e9185650216f3c3fa97759caf67ee79db66cb5e1)
```

`lsb_release -a`:
```
LSB Version:	core-9.20160110ubuntu0.2-amd64:core-9.20160110ubuntu0.2-noarch:security-9.20160110ubuntu0.2-amd64:security-9.20160110ubuntu0.2-noarch
Distributor ID:	LinuxMint
Description:	Linux Mint 18 Sarah
Release:	18
Codename:	sarah
```

**Additional context**
A situation where this issue is relevant occurs in the Java standard library in ``java.util.ServiceLoader``:

```
...
 * <pre>{@code
 *     ServiceLoader<CodecFactory> loader = ServiceLoader.load(CodecFactory.class);
 *     Set<CodecFactory> pngFactories = loader
 *            .stream()                                              // Note a below
 *            .filter(p -> p.type().isAnnotationPresent(PNG.class))  // Note b
 *            .map(Provider::get)                                    // Note c
 *            .collect(Collectors.toSet());
 * }</pre>
 * <ol type="a">
 *   <li> A stream of {@code Provider<CodecFactory>} objects </li>
 *   <li> {@code p.type()} yields a {@code Class<CodecFactory>} </li>
 *   <li> {@code get()} yields an instance of {@code CodecFactory} </li>
 * </ol>
...
```

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8523

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by Stewori** [comment]
Cool. Thanks!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
